pkg/api/handlers/compat/images_build.go

```diff
@@ -53,7 +61,7 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 
 	anchorDir, err := os.MkdirTemp("", "libpod_builder")
 	if err != nil {
-		utils.InternalServerError(w, err)
+		utils.InternalServerError(w, genSpaceErr(err))
```

Comment:
This doesn't seem right here. We create a tmp directory on the server here so if this ever is out of space here then the server is out of space independently of the context dir size so we should not say it for this condition. IF anything it must say server is out of space.

Reply:
These are the errors the user is seeing when they are out of space.
mkdir /var/tmp/libpod_builderXXXXX

Reply:
jiazha-mac:~ jiazha$ podman build -t quay.io/olmqe/etcd-index:fips -f catalog.Dockerfile 
Error: mkdir /var/tmp/libpod_builder3448793565: no space left on device

Reply:
yes correct because the server is out of space thus saying the context dir is to big makes no sense at all because we didn't even started copying the tarball to the fs at this point.

The actual copy happens in extractTarFile() so there the error annotation makes sense

Reply:
I agree, and  yet this is the error the user is seeing and it is fixed by using a different context dir.

Reply:
because the first test was done on older versions seems to have leaked the dir

```
Error: write /var/tmp/libpod_builder1195209327/build/.local/share/containers/podman/machine/applehv/podman-machine-default-arm64.raw: no space left on device
``` 
This is the second error reported which is right because here it fails at the big VM image it tried to copy into the VM image...